File: lib/hdbfs/defs.py
import logging
import os

import hdbfs.model as model

VERSION = 1
REVISION = 0

DB_VERSION  = model.VERSION
DB_REVISION = model.REVISION

DEFAULT_LIBRARY = os.path.join( os.environ['HOME'], '.higu' )
HIGURASHI_DB_NAME = 'hfdb.dat'

from hdbfs.model import ObjectClass, ObjectType

NAME_POLICY_DONT_REGISTER   = 0
NAME_POLICY_DONT_SET        = 1
NAME_POLICY_SET_IF_UNDEF    = 2
NAME_POLICY_SET_ALWAYS      = 3

LOG = logging.getLogger( __name__ )

File: lib/src/main/kotlin/ca/_4haven/higu/hdbfs/defs.kt
package ca._4haven.higu.hdbfs

val VERSION = 2
val REVISION = 0

val HIGURASHI_DB_NAME = "hfdb.dat"

val TYPE_NILL       = 0
val TYPE_FILE       = 1000
val TYPE_GROUP      = 2000
val TYPE_ALBUM      = 2001
val TYPE_CLASSIFIER = 2002

val SP_EXPENDABLE = 1000
val SP_NORMAL     = 2000
val SP_PRIORITY   = 3000

val NAME_POLICY_DONT_REGISTER   = 0
val NAME_POLICY_DONT_SET        = 1
val NAME_POLICY_SET_IF_UNDEF    = 2
val NAME_POLICY_SET_ALWAYS      = 3

typealias NamePolicy = Int
